TimoCloud-Universal/src/main/java/cloud/timo/TimoCloud/cord/sockets/CordStringHandler.java

```diff
@@ -1,54 +1,32 @@
-import cloud.timo.TimoCloud.api.implementations.managers.EventManager;
-import cloud.timo.TimoCloud.api.messages.objects.AddressedPluginMessage;
-import cloud.timo.TimoCloud.api.utils.EventUtil;
 import cloud.timo.TimoCloud.common.protocol.Message;
+import cloud.timo.TimoCloud.common.protocol.MessageType;
 import cloud.timo.TimoCloud.common.sockets.BasicStringHandler;
-import cloud.timo.TimoCloud.common.utils.EnumUtil;
-import cloud.timo.TimoCloud.common.utils.PluginMessageSerializer;
-import cloud.timo.TimoCloud.cord.TimoCloudCord;
-import cloud.timo.TimoCloud.cord.api.TimoCloudUniversalAPICordImplementation;
+import cloud.timo.TimoCloud.cord.sockets.handler.CordApiDataHandler;
+import cloud.timo.TimoCloud.cord.sockets.handler.CordEventFiredHandler;
+import cloud.timo.TimoCloud.cord.sockets.handler.CordHandshakeSuccessHandler;
+import cloud.timo.TimoCloud.cord.sockets.handler.CordPluginMessageHandler;
 import io.netty.channel.Channel;
 import io.netty.channel.ChannelHandler;
 
-import java.util.Map;
-
 @ChannelHandler.Sharable
 public class CordStringHandler extends BasicStringHandler {
 
+    public CordStringHandler() {
+        addBasicHandlers();
+    }
+
     @Override
     public void handleMessage(Message message, String originalMessage, Channel channel) {
-        String type = (String) message.get("type");
-        Object data = message.get("data");
-        switch (type) {
-            case "HANDSHAKE_SUCCESS":
-                TimoCloudCord.getInstance().onHandshakeSuccess();
-                break;
-            case "API_DATA": {
-                ((TimoCloudUniversalAPICordImplementation) TimoCloudAPI.getUniversalAPI()).setData((Map<String, Object>) data);
-                break;
-            }
-            case "EVENT_FIRED":
-                try {
-                    EventType eventType = EnumUtil.valueOf(EventType.class, (String) message.get("eT"));
-                    ((EventManager) TimoCloudAPI.getEventAPI()).callEvent(((TimoCloudUniversalAPIBasicImplementation) TimoCloudAPI.getUniversalAPI()).getObjectMapper().readValue((String) data, EventUtil.getClassByEventType(eventType)));
-                } catch (Exception e) {
-                    System.err.println("Error while parsing event from json: ");
-                    TimoCloudCord.getInstance().severe(e);
-                }
-                break;
-            case "PLUGIN_MESSAGE": {
-                AddressedPluginMessage addressedPluginMessage = PluginMessageSerializer.deserialize((Map) data);
-                ((TimoCloudMessageAPIBasicImplementation) TimoCloudAPI.getMessageAPI()).onMessage(addressedPluginMessage);
-                break;
-            }
-            default:
-                TimoCloudCord.getInstance().severe("Could not categorize json message: " + originalMessage);
-        }
+        MessageType type = message.getType();
+        getMessageHandlers(type).forEach(messageHandler -> messageHandler.execute(message, channel));
```

Comment:
We need some kind of exception handling so that the execution does not stop if one of the handlers throws an exception